frontend/src/components/namespace/CreateNamespaceButton.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+} from '@mui/material';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+import { post } from '../../lib/k8s/apiProxy';
+import Namespace from '../../lib/k8s/namespace';
+import { clusterAction } from '../../redux/clusterActionSlice';
+import { EventStatus, HeadlampEventType, useEventCallback } from '../../redux/headlampEventSlice';
+import { ActionButton, AuthVisible } from '../common';
+
+export function CreateNamespaceButton() {
+  const { t } = useTranslation(['glossary', 'translation']);
+  const [namespaceName, setNamespaceName] = useState('');
+  const [namespaceDialogOpen, setNamespaceDialogOpen] = useState(false);
+  const dispatchCreateEvent = useEventCallback(HeadlampEventType.CREATE_RESOURCE);
+  const location = useLocation();
+  const dispatch = useDispatch();
+
+  function createNewNamespace() {
+    async function namespaceRequest() {
+      const newNamespaceData = {
+        apiVersion: 'v1',
+        kind: 'Namespace',
+        metadata: {
+          name: namespaceName,
+        },
+      };
+      try {
+        const response = await post('/api/v1/namespaces', newNamespaceData);
+        return response;
+      } catch (error) {
+        console.error('Error creating namespace', error);
+        throw error;
+      }
+    }
+
+    setNamespaceDialogOpen(false);
+    dispatch(
+      clusterAction(() => namespaceRequest(), {
+        startMessage: t('Creating namespace {{ itemName }}…', { namespaceName }),
+        cancelledMessage: t('Cancelled creation of {{ itemName }}.', { namespaceName }),
+        successMessage: t('Created Namespace {{ itemName }}.', { namespaceName }),
+        errorMessage: t('Error creating namespace {{ itemName }}.', { namespaceName }),
```

Comment:
The `itemName` doesn't match any key in the passed object. And testing shows:
![Screenshot showing the cancel message without the name of the namespace](https://github.com/headlamp-k8s/headlamp/assets/1029635/3a54edb0-6bc7-4a7f-9f52-c3032fb8cf9a)

Reply:
fixed! let me know if this usage is okay, since the variable being used a custom name by the user not sure if it would need a direct translation back to them